### PR TITLE
feat: Add support for 01 model platform

### PR DESCRIPTION
**Pull request by MuggleJinx**
## Description

This PR adds support for the [Yi-series LLM models](https://platform.lingyiwanwu.com/), including yi-lightning, yi-large, yi-medium, and yi-large-turbo. The addition of Yi-series models enhances the platform’s capabilities, providing support for a broader range of language models and enabling users to leverage different performance tiers.

## Motivation and Context

Closes issue #1066.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [x] Implement consistent interface for Yi-series model.
- [x] Add the corresponding example, and test file. 

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/configs/yi_config.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        frequency_penalty (float, optional): Number between :obj:`-2.0` and
+            :obj:`2.0`. Positive values penalize new tokens based on their
+            existing frequency in the text so far, decreasing the model's
+            likelihood to repeat the same line verbatim. (default: :obj:`0.0`)
+        logit_bias (dict, optional): Modify the likelihood of specified tokens
+            appearing in the completion. Accepts a json object that maps tokens
+            (specified by their token ID in the tokenizer) to an associated
+            bias value from :obj:`-100` to :obj:`100`. (default: :obj:`{}`)
+        user (str, optional): A unique identifier representing your end-user,
+            which can help monitor and detect abuse. (default: :obj:`""`)
```
Comment (Wendong-Fan) [inline camel/configs/yi_config.py:67]: from yi's doc, I didn't see they mentioned they support these arguments, could you give me some reference?

File: camel/types/enums.py [hunk 2]
```diff
@@ -87,6 +87,13 @@ class ModelType(UnifiedModelType, Enum):
     REKA_FLASH = "reka-flash"
     REKA_EDGE = "reka-edge"
 
+    # Yi models (01-ai)
+    YI_LIGHTNING = "yi-lightning"
+    YI_LARGE = "yi-large"
+    YI_MEDIUM = "yi-medium"
+    YI_LARGE_TURBO = "yi-large-turbo"
```
Comment (Wendong-Fan) [inline camel/types/enums.py:94]: from https://platform.lingyiwanwu.com/docs#%E6%A8%A1%E5%9E%8B%E4%B8%8E%E8%AE%A1%E8%B4%B9 there's more models supported by Yi, could we include all?

File: camel/utils/token_counting.py [hunk 3]
```diff
@@ -405,3 +405,28 @@ def _convert_response_from_openai_to_mistral(
...
+# The API does not provide official token counting for Yi models, using the default OpenAI tokenizer.
+class YiTokenCounter(BaseTokenCounter):
+    def __init__(self, model_type: UnifiedModelType):
+        r"""Constructor for the token counter for Yi models.
+
+        Args:
+            model_type (UnifiedModelType): Model type for which tokens will be
+                counted.
+        """
+        self._internal_tokenizer = OpenAITokenCounter(model_type)
```
Comment (Wendong-Fan) [inline camel/utils/token_counting.py:419]: how could Yi's `model_type` used by `OpenAITokenCounter`?

**Review by Wendong-Fan — COMMENTED** [review]
thanks @MuggleJinx 's contribution! Left some comments below

File: camel/configs/yi_config.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            values increase randomness. (default: :obj:`0.9`)
+        temperature (float, optional): Controls the diversity and focus of
+            the generated results. Lower values make the output more focused,
+            while higher values make it more diverse. (default: :obj:`0.3`)
+        stream (bool, optional): If True, enables streaming output.
+            (default: :obj:`False`)
+    """
+
+    tool_choice: Optional[Union[dict[str, str], str]] = None
+    max_tokens: Optional[int] = 5000
```
Comment (Wendong-Fan) [inline camel/configs/yi_config.py:51]: don't limit max_token from our side, set it to `NOT_GIVEN`

```from camel.types import NOT_GIVEN```

```suggestion
    max_tokens: Optional[int] = NOT_GIVEN
```

File: camel/models/model_factory.py [hunk 5]
```diff
@@ -108,6 +109,8 @@ def create(
             model_class = RekaModel
         elif model_type == ModelType.STUB:
             model_class = StubModel
+        elif model_platform.is_yi and model_type.is_yi:
+            model_class = YiModel
```
Comment (lightaime) [inline camel/models/model_factory.py:113]: Would be better to move this above the `ModelType.STUB`.

**Pull request merged**